### ISSUE TITLE
Implement $node->setPrimaryType

### ISSUE THREAD

**Issue by dantleech** [jackalope/jackalope#247]
Is there a reason why `setPrimaryType` is not supported?

```
    public function setPrimaryType($nodeTypeName)
    {
        $this->checkState();

        throw new NotImplementedException('Write');
    }
```

**Comment by danrot** [comment]
As mentioned in https://github.com/phpcr/phpcr-api-tests/pull/160#discussion_r32242150 this change would also have to be applied to the verisoning, since the `frozenPrimaryType` of the frozen node then has to be copied to the actual node.

**Comment by dbu** [comment]
good point. but we don't need that for your versioning implementation. we just need to make sure when we implement the feature that we also implement restoring.
